src/main/java/io/jenkins/plugins/kubernetes/NoDelayProvisionerStrategy.java

```diff
@@ -64,26 +65,29 @@
                         continue;
                     }
                 }
+                canProvision = true;
                 Collection<NodeProvisioner.PlannedNode> plannedNodes = cloud.provision(label, workloadToProvision);
                 LOGGER.log(Level.FINE, "Planned {0} new nodes", plannedNodes.size());
                 fireOnStarted(cloud, strategyState.getLabel(), plannedNodes);
                 strategyState.recordPendingLaunches(plannedNodes);
                 availableCapacity += plannedNodes.size();
+                totalPlannedNodes += plannedNodes.size();
                 LOGGER.log(Level.FINE, "After provisioning, available capacity={0}, currentDemand={1}", new Object[]{availableCapacity, currentDemand});
                 break;
             }
         }
-        if (availableCapacity > previousCapacity && label != null) {
-            LOGGER.log(Level.FINE, "Suggesting NodeProvisioner review");
-            Timer.get().schedule(label.nodeProvisioner::suggestReviewNow, 1L, TimeUnit.SECONDS);
-        }
-        if (availableCapacity >= currentDemand) {
-            LOGGER.log(Level.FINE, "Provisioning completed");
-            return NodeProvisioner.StrategyDecision.PROVISIONING_COMPLETED;
+        if (currentDemand - availableCapacity <= 0) {
+            LOGGER.log(Level.FINE, String.format("Provisioning completed for label: [%s]", label));
         } else {
-            LOGGER.log(Level.FINE, "Provisioning not complete, consulting remaining strategies");
-            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+            if (!canProvision) {
+                return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+            }
```

Comment:
So the key change resides here. If the Kubernetes strategy did not provision any node through `KubernetesCloud`, it goes to the next strategy, allowing other clouds to pick up the request.

However as soon as it was able to provision at least 1 node using a `KubernetesCloud`, we consider this label isn't served by other clouds so we just end the provisioning to prevent the standard strategy from causing overprovisioning.

Reply:
Thanks for noting! I should have documented this in the PR description but I am not very familiar with Jenkins provisioning logic.

If a label is served by multiple strategies, and the Kubernetes strategy serves the first N requests up to the concurrency limit, will the remaining requests _eventually_ be served by other strategies?

Reply:
Not in the current round. If we reach the limit, on the next round, the Kubernetes strategy will provision 0 node and pass the ball to other strategies, however there won't be any more in-flight node because they get processed before each round of strategies (and the Kubernetes one returns instantly)